### ISSUE TITLE
cv2.error: !ssize.empty() in function 'resize'

### ISSUE THREAD

**Issue by ew-kim** [shirowalker/UCAD#4]
Thank you for your great work, I have a question.
When I try the vision_transformer.py code, I get the following error. Can you tell me how to solve it?

 ```
for i in range(x.shape[0]):
                if('mvtec2d' in image_path[i]):
                    sam_score = cv2.imread(image_path[i].replace('mvtec2d','mvtec2d-sam-b'))
                elif('visa' in image_path[i]):
                    sam_score = cv2.imread(image_path[i].replace('visa','visa-sam-b'))
                labels[i] = torch.from_numpy(cv2.resize(sam_score,(14,14))[:,:,0].flatten()).cuda()
            res['loss'] = torch.tensor(0).float().cuda()
            # loss for sam
            for k in range(len(res['seg_feat'])):
                res['loss'] += self.contrastive_loss(res['seg_feat'][k], labels, temperature=0.5)
        else:
            pass
        
        return res
```

![image](https://github.com/shirowalker/UCAD/assets/100407027/d8fa2ecd-3cfa-4ed7-b17d-1ddf5feee97a)
>>File "/home/ewkim/Desktop/wan/UCAD/UCAD-main/patchcore/vision_transformer.py", line 713, in forward_head
    labels[i] = torch.from_numpy(cv2.resize(sam_score,(14,14))[:,0,0].flatten()).cuda()
cv2.error: OpenCV(4.6.0) /io/opencv/modules/imgproc/src/resize.cpp:4052: error: (-215:Assertion failed) !ssize.empty() in function 'resize'

**Comment by shirowalker** [comment]
> Thank you for your great work, I have a question. When I try the vision_transformer.py code, I get the following error. Can you tell me how to solve it?
> 
> ```
> for i in range(x.shape[0]):
>                if('mvtec2d' in image_path[i]):
>                    sam_score = cv2.imread(image_path[i].replace('mvtec2d','mvtec2d-sam-b'))
>                elif('visa' in image_path[i]):
>                    sam_score = cv2.imread(image_path[i].replace('visa','visa-sam-b'))
>                labels[i] = torch.from_numpy(cv2.resize(sam_score,(14,14))[:,:,0].flatten()).cuda()
>            res['loss'] = torch.tensor(0).float().cuda()
>            # loss for sam
>            for k in range(len(res['seg_feat'])):
>                res['loss'] += self.contrastive_loss(res['seg_feat'][k], labels, temperature=0.5)
>        else:
>            pass
>        
>        return res
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/100407027/323119461-d8fa2ecd-3cfa-4ed7-b17d-1ddf5feee97a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzNDAwMDgsIm5iZiI6MTcxMzMzOTcwOCwicGF0aCI6Ii8xMDA0MDcwMjcvMzIzMTE5NDYxLWQ4ZmEyZWNkLTNjZmEtNGVkNy1iMTdkLTFkZGY1ZmVlZTk3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxN1QwNzQxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTQzZWE0YmNjYzUxNzdkMWU2YzBhZGNkM2RhN2ExMDA5ODNhZDYwNzhkZjdjMzY0OTYyZjQ0YjllYzljYzFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.bjJa-rdsX5g12L63XGvuaYCEEBagXbINvoibcSf-pHA)
> 
> > > File "/home/ewkim/Desktop/wan/UCAD/UCAD-main/patchcore/vision_transformer.py", line 713, in forward_head
> > > labels[i] = torch.from_numpy(cv2.resize(sam_score,(14,14))[:,0,0].flatten()).cuda()
> > > cv2.error: OpenCV(4.6.0) /io/opencv/modules/imgproc/src/resize.cpp:4052: error: (-215:Assertion failed) !ssize.empty() in function 'resize'

As shown in the code, there needs to be 'mvtec2d-sam-b' in the same directory as 'mvtec2d'. 'mvtec2d' is the original data set, and 'mvtec2d-sam-b' is the file processed by SAM, which has also been provided in the code repository. The code error is most likely due to the two image folders not being placed correctly.
